services/user.py

```diff
@@ -0,0 +1,56 @@
+from django.contrib.auth import get_user_model
+
+from db.models import User
+
+
+def create_user(
+    username: str,
+    password: str,
+    email: str = None,
+    first_name: str = None,
+    last_name: str = None
+) -> User:
+    user = get_user_model().objects.create_user(username=username)
+    user.set_password(password)
+
+    if email:
+        user.email = email
+    if first_name:
+        user.first_name = first_name
+    if last_name:
+        user.last_name = last_name
```

Comment:
You're calling `save()` method once, which is good. But you can optimize your code by creating a dictionary with user data and then unpacking it in `create_user` method. This way, you can avoid multiple attribute assignments and make your code cleaner.